src/Symfony/Component/HtmlSanitizer/HtmlSanitizerConfig.php

```diff
@@ -0,0 +1,349 @@
+     */
+    public function withAllowedLinksHosts(?array $allowedLinksHosts): self
+    {
+        $self = clone $this;
+        $self->allowedLinksHosts = $allowedLinksHosts;
+
+        return $self;
+    }
+
+    /**
+     * Configure the sanitizer to retain relative links.
+     */
+    public function withAllowedLinksRelative(bool $allowedLinksRelative = true): self
+    {
+        $self = clone $this;
+        $self->allowedLinksRelative = $allowedLinksRelative;
+
+        return $self;
+    }
+
+    /**
+     * Configure the URL schemes to be retained in href attributes of media elements (img, audio and video).
+     * All other schemes will be dropped as they can lead to script execution.
+     *
+     * Calling this method will override all previous configuration for allowed media schemes.
+     */
+    public function withAllowedMediaSchemes(array $allowedMediaSchemes): self
+    {
+        $self = clone $this;
+        $self->allowedMediaSchemes = $allowedMediaSchemes;
+
+        return $self;
+    }
+
+    /**
+     * Configure the URL hosts to be retained in href attributes of media elements (img, audio and video).
+     * All other hosts will be dropped. By default all hosts are allowed ($allowedMediaHosts = null).
+     *
+     * Calling this method will override all previous configuration for allowed links schemes.
+     */
+    public function withAllowedMediaHosts(?array $allowedMediaHosts): self
+    {
+        $self = clone $this;
+        $self->allowedMediaHosts = $allowedMediaHosts;
+
+        return $self;
+    }
+
+    /**
+     * Configure the sanitizer to retain relative media URL.
+     */
+    public function withAllowedMediaRelative(bool $allowedMediaRelative = true): self
+    {
+        $self = clone $this;
+        $self->allowedMediaRelative = $allowedMediaRelative;
+
+        return $self;
+    }
+
+    /**
+     * Configure the sanitizer to transform URLs using the HTTP scheme to use the HTTPS scheme instead.
+     */
+    public function forceHttpsUrls(bool $forceHttpsUrls = true): self
+    {
+        $self = clone $this;
+        $self->forceHttpsUrls = $forceHttpsUrls;
+
+        return $self;
+    }
+
+    /**
+     * Configure the given element as allowed. Allowed elements are elements the sanitizer
+     * should retain from the input.
+     *
+     * A list of allowed attributes for this element can be passed as a second argument.
+     * Passing "*" will allow all standard attributes on this element. By default, no
+     * attributes are allowed on the element.
+     *
+     * Calling this method will override all previous configuration for the given element.
+     */
+    public function allowElement(string $element, array|string $allowedAttributes = []): self
+    {
+        $self = clone $this;
+
+        // Unblock the element is necessary
+        if (isset($self->blockedElements[$element])) {
+            unset($self->blockedElements[$element]);
+        }
```

Comment:
```suggestion
        unset($self->blockedElements[$element]);
```

`unset()` should be idempotent. No need to safe-guard the call.